qutebrowser/browser/webengine/spell.py

```diff
@@ -0,0 +1,150 @@
+        """Create Language object from a tuple.
+
+        Args:
+            lang_tuple: a tuple of strings containing
+                        the specs of the language in the following format:
+                        (code, name, file)
+        """
+        code, name, file = lang_tuple
+        return Language(code, name, file)
+
+    @staticmethod
+    def from_tsv_string(tsv_string):
+        """Create Language object from a string in tab-separated values format.
+
+        Args:
+            tsv_string: a string containing
+                        the specs of the language in the following format:
+                        "code   name    file"
+        """
+        lang_array = tsv_string.split('\t')
+        return Language.from_array(lang_array)
+
+    def __repr__(self):
+        return 'Language({}, {}, {})'.format(self.code, self.name, self.file)
+
+
+def get_dictionary_dir():
+    """Return the path to the QtWebEngine's dictionaries directory."""
+    return os.path.join(QLibraryInfo.location(QLibraryInfo.DataPath),
+                        'qtwebengine_dictionaries')
+
+
+def get_language_list_file():
+    """Return the path to the file with the list of all available languages."""
+    package_dir = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(package_dir, 'langs.tsv')
+
+
+def get_available_languages():
+    """Return a list of Language objects of all available languages."""
+    with open(get_language_list_file(), 'r', encoding='UTF-8') as file:
+        return [Language.from_tsv_string(line[:-1]) for line in file]
+
+
+def get_installed_languages():
+    """Return a list of Language objects of all installed languages."""
+    if not os.path.isdir(get_dictionary_dir()):
+        return []
+    installed_files = [os.path.basename(file)
+                       for file in os.listdir(get_dictionary_dir())]
+    all_languages = get_available_languages()
+    return filter_languages(all_languages, installed_files,
+                            by=lambda lang: lang.file,
+                            fail_on_unknown=False)
+
+
+def filter_languages(languages, selected, by=lambda lang: lang.code,
+                     fail_on_unknown=True):
+    """Filter a list of languages based on an inclusion list.
+
+    Args:
+        languages: a list of languages to filter
+        selected: a list of keys to select
+        by: a function returning the selection key (code by default)
+        fail_on_unknown: whether to raise an error if there is an unknown
+                         key in selected
+    """
+    filtered_languages = []
+    for language in languages:
+        if by(language) in selected:
+            filtered_languages.append(language)
+            selected.remove(by(language))
+    if fail_on_unknown and selected:
+        unknown = ', '.join(selected)
+        raise ValueError('unknown languages found: {}'.format(unknown))
+    return filtered_languages
+
+
+def download_dictionary(url, dest):
```

Comment:
This and `install` look like they'd fit better in `scripts/install_dict.py` instead of here, since they're intended for that script and not usage inside qutebrowser.